tests/local/custom-page-sizes/custom-page-sizes-invalid.wast

```diff
@@ -0,0 +1,12 @@
+(assert_malformed
+ (module binary
+   "\00asm" "\01\00\00\00"
+   "\05\04\01"                ;; Memory section
+
+   ;; memory 0
+   "\08"                      ;; flags w/ custom page size
+   "\00"                      ;; minimum = 0
+   "\11"                      ;; pagesize = 2**17
+ )
+ "invalid custom page size"
+)
```

Comment:
Can this be rewritten as `*.wast` now that it's no longer an error there?

Also mind adding a test where the page size is >=17, e.g. `1<<64`?

Reply:
Also can you add a test for `(pagesize 3)` for a non-power-of-two test?

Reply:
> Also can you add a test for `(pagesize 3)` for a non-power-of-two test?

We can't do that because we still need to check that the page size is a power of two in the wast parser because we cannot delay the error until encoding time because the wast encoding methods are infallible but because we can't losslessly turn a non-power-of-two into its ilog2 we would need to be able to fail at encoding time. So instead we still eagerly check for powers of two at parse time, which means that the whole `.wast` test file gets rejected as invalid text format rather than the particular module being invalid and passing the `assert_invalid` directive.

Reply:
Oh sorry, I mean to use `assert_malformed` for this, not `assert_invalid`, which specifically tests text-level errors

Reply:
Even with `assert_malformed` instead of `assert_invalid`:

```
failed test: tests/local/custom-page-sizes/custom-page-sizes-invalid.wast

Caused by:
    0: failed to run `json-from-wast` cli subcommand
    1: failed to run "/home/nick/wasm-tools/target/debug/wasm-tools" "json-from-wast" "tests/local/custom-page-sizes/custom-page-sizes-invalid.wast" "--pretty" "--wasm-dir" "/tmp/.tmphI5u29"
       stdout: 
       stderr: error: invalid custom page size: 3
            --> tests/local/custom-page-sizes/custom-page-sizes-invalid.wast:2:31
             |
           2 |   (module (memory 0 (pagesize 3)))
             |                               ^
```

Reply:
I don't think we can resolve this without making `wast`'s encoding fallible, which doesn't seem worth it.

Reply:
oh nice!